src/shmem_team.c

```diff
@@ -187,6 +193,7 @@ void shmem_internal_team_fini(void)
     free(shmem_internal_team_pool);
     shmem_internal_free(shmem_internal_psync_pool);
     shmem_internal_free(psync_pool_avail);
+    shmem_internal_free(team_ret_val);
```

Comment:
Given the above NULL check comment, you may want to guard the above free with `if (team_ret_val != NULL)`...

Reply:
Doesn't `shmem_internal_free` already have a NULL check?

Reply:
So it does, but actually this function won't get called if init fails (see below) so we need to handle cleanup before returning from init.